tests/shell/shell_recorder/shell_recorder.sh

```diff
@@ -1,12 +1,24 @@
 #!/bin/bash
 
+getTempDir()
+{
+	if [ -z "$TMPDIR" ]; then
+		TMPDIR="/tmp"
+	else
+		if [[ "$TMPDIR" == */ ]]; then
+			TMPDIR=$(echo $TMPDIR | head -c -2)
+		fi
+	fi
+	echo "$TMPDIR"
+}
+
 FILE=$1
 Mountpoint=
 DBFile=
 Storage=
 MountArgs=
 DiffType=File
-OutFile=$(mktemp "${TMPDIR:-/tmp}/tmp.XXXXXXXXX")
+OutFile=$(mktemp "$(getTempDir)/tmp.XXXXXXXXX")
```

Comment:
> not sure, I use now
`mktemp -t elektraenv.XXXXXXXXX 2>/dev/null || mktemp -t 'elektraenv'`
which file will be created with this command?

The command `mktemp -t elektraenv.XXXXXXXXX 2>/dev/null` creates something like:
```
/var/folders/hx/flbncdhj4fs87095gzxvnj3h0000gn/T/elektraenv.XXXXXXXXX.sTrHgctr
```
, while `mktemp -t 'elektraenv'` creates something like

```
/var/folders/hx/flbncdhj4fs87095gzxvnj3h0000gn/T/elektraenv.h0Zg7gVB
```

The overall result will be the first output, since `mktemp -t 'elektraenv'` works without problems.